web/packages/teleport/src/components/Wizard/flow.tsx

```diff
@@ -17,24 +17,37 @@
  */
 
 export type BaseView<T> = T & {
+  // whether to hide the view from the list of views
   hide?: boolean;
+  // current step index in the wizard
   index?: number;
+  // current visible step index in the wizard (ignoring hidden steps)
+  displayIndex?: number;
   views?: BaseView<T>[];
   title: string;
 };
 
 /**
  * computeViewChildrenSize calculates how many children a view has, without counting the first
  * child. This is because the first child shares the same index with its parent, so we don't
- * need to count it as it's not taking up a new index
+ * need to count it as it's not taking up a new index.
+ *
+ * If `constrainToVisible` is true, then we only count the visible views.
  */
-export function computeViewChildrenSize<T>(views: BaseView<T>[]) {
+export function computeViewChildrenSize<T>(
+  views: BaseView<T>[],
+  constrainToVisible = false
```

Comment:
It might be a good idea to convert this arg to a named argument to avoid callsites like `computeViewChildrenSize(view.views, true)` on line 86, where it's not clear just from reading the callsite what `true` does.

```suggestion
  { constrainToVisible = false } = {}
```